From 734a51ad634a2e6954529b88d2e1febdede3eac8 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Wed, 31 Mar 2021 12:53:07 +0300
Subject: [PATCH] [SW-36011] gaudi: erase fit from spsram

Erase fit contents from SP SPRAM after using
it, as a security enhancement.

Change-Id: Ic18a21506ecb2a1df3a74081aa262335a79c6310
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index 168fe5d..15f06bb 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -23,6 +23,7 @@
 #define TIMEOUT_US		1000
 
 static bool is_fit_checked;
+static unsigned int map_size;
 
 uintptr_t fit_base_offset;
 
@@ -109,6 +110,8 @@
 		goto fwupd_map_err;
 	}
 
+	map_size = size;
+
 	if (cmd == HL_UPD_MAP_SRAM)
 		preboot_security_dynamic_img_sram_open();
 	else
@@ -191,6 +194,10 @@
 		preboot_security_dynamic_img_spsram_close();
 
 	err = module_fwupd_load_type();
+
+	if (cmd == HL_UPD_RUN_SPSRAM)
+		memset((void *)fit_base_offset, 0, ROUND_UP4(map_size));
+
 	if (err) {
 		sts = HL_UPD_STS_VALID_ERR;
 		preboot_puts("Failed to load FIT image\n");
